src/autolabel/labeler.py

```diff
@@ -478,6 +480,32 @@ def generate_explanations(
 
         return seed_examples
 
+    def generate_synthetic_dataset(
+        self, examples_per_label: int = 5, guidelines: str = None
+    ) -> AutolabelDataset:
+        columns = get_format_variables(self.config.example_template())
+        df = pd.DataFrame(columns=columns)
+        for label in self.config.labels_list():
+            prompt = self.task.get_generate_dataset_prompt(
+                label, examples_per_label, guidelines
```

Comment:
instead of passing the guidelines as a string input, we should pass this through the config to keep this in line with how we accept prompts for labeling.

lets accept examples per label there as well